backend/src/test/java/de/neuefische/ffmjava221/teamprojekt/backend/meals/MealIntegrationTest.java

```diff
@@ -144,4 +144,11 @@ void deleteMealSuccesfullAndReturnMealToDelete() throws Exception {
                     }
                     """.replace("<id>",meal.id())));
     }
+
+    @Test
+    @DirtiesContext
+    void deleteMealWithWrongIdReturns400() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.delete("/api/meals/123"))
+                .andExpect(status().is(400));
+    }
```

Comment:
you test here of status 400

we tested it with exception there is the status code 500


"timestamp": "2022-10-27T14:07:13.417+00:00",
    "status": 500,
    "error": "Internal Server Error",
    "trace": "java.util.NoSuchElementException: NoSuchEmployeeFound

hm you think 400 is ok ?

Reply:
Test with postman on your meal url with maybe 123

Reply:
We get Status 400 because our Controller responds with 400 if there is no ID found:
```
@PutMapping("/{id}")
    public ResponseEntity<Meal> updateMeal(@PathVariable String id, @RequestBody Meal meal){
        if (meal.id().equals(id)) {
            int index = mealService.getIndexOfId(id);
                    Meal newMeal = mealService.updateMeal(index, meal);
                    boolean mealExists = (index>=0);
                    return mealExists ? new ResponseEntity<>(newMeal,HttpStatus.OK) : new ResponseEntity<>(newMeal,HttpStatus.CREATED);
        } return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
    }
```